test/unit/org/apache/cassandra/utils/RecomputingSupplierTest.java

```diff
@@ -93,31 +89,40 @@ public void recomputingSupplierTest() throws Throwable
                             lastSeen.accumulateAndGet(v, Math::max);
                             Assert.assertTrue(String.format("Seen %d out of order. Last seen value %d", v, seenBeforeGet),
                                               v >= seenBeforeGet);
-
                         }
-
                     }
                     catch (Throwable e)
                     {
                         thrown.set(e);
-                        latch.countDown();
                     }
                 }
             });
         }
 
-        latch.await(10, TimeUnit.SECONDS);
-
+        Util.spinAssertEquals(true, () -> {return counter.get() > 1;}, 11);
         testExecutor.shutdown();
-        testExecutor.awaitTermination(10, TimeUnit.SECONDS);
-        executor.shutdown();
-        executor.awaitTermination(10, TimeUnit.SECONDS);
+        Assert.assertTrue(testExecutor.awaitTermination(11, TimeUnit.SECONDS));
```

Comment:
Why are we waiting 11 seconds in the calls to `spinAssertEquals` and `awaitTermination`? If it's an arbitrary amount of time we could use a round number like for example 10s.

Reply:
Part of a left over from a previous approach where I had 10 threads sleeping 1s, hence I was waiting 11s. We can change it to 10 which reads better.